File: javascript/lib/elements/missing/missing_flag.js
import { MissingElementBase } from './missing_element_base.js';

export class MissingFlag extends MissingElementBase {
    get [Symbol.toStringTag]() {
        return 'MissingFlag';
    }
    
    toString() {
        if (this._key === null)
            return `[object MissingFlag]`;
        
        return `[object MissingFlag key=${this._key}]`;
    }
}

File: ruby/lib/enolib/elements/missing/missing_flag.rb
# frozen_string_literal: true

module Enolib
  class MissingFlag < MissingElementBase
    def to_s
      if @key
        "#<Enolib::MissingFlag key=#{@key}>"
      else
        '#<Enolib::MissingFlag>'
      end
    end
  end
end
